packages/matter.js/src/CommissioningServer.ts

```diff
@@ -370,6 +385,81 @@ export class CommissioningServer extends MatterNode {
         await this.deviceInstance.start();
     }
 
+    updateStructure() {
+        logger.debug("Endpoint structure got updated ...");
+        this.ensureEndpointIds(); // Make sure to have unique endpoint ids
+        this.interactionServer?.setRootEndpoint(this.rootEndpoint); // Reinitilize the interaction server structure
+    }
+
+    getNextEndpointId(increase = true) {
+        if (increase) {
+            this.nextEndpointId++;
+        }
+        return this.nextEndpointId;
+    }
+
+    ensureEndpointIds() {
+        const rootUniqueIdPrefix = this.rootEndpoint.determineUniqueID();
+        this.initializeEndpointIdsFromStorage(this.rootEndpoint, rootUniqueIdPrefix);
+        this.fillAndStoreEndpointIds(this.rootEndpoint, rootUniqueIdPrefix);
+        this.endpointStructureStorage?.set("nextEndpointId", this.nextEndpointId);
+    }
+
+    private initializeEndpointIdsFromStorage(endpoint: Endpoint, parentUniquePrefix = "") {
+        if (this.endpointStructureStorage === undefined) {
+            throw new Error("endpointStructureStorage not set!");
```

Comment:
Changed to "Storage manager must be initialized to enable initialization from storage." ... because the endpointStructureStorage is automatically initialized with the storage manager